src/alpaka/AlpakaCore/alpakaWorkDivHelper.h

```diff
@@ -8,234 +8,225 @@ using namespace alpaka_common;
-        endElementIdxUncutVec[dimIndex] = endElementIdxUncut;
-      }
+    template <typename TAcc>
+    ALPAKA_FN_ACC std::pair<Idx, Idx> element_index_range_in_block(const TAcc& acc,
+                                                                   const Idx elementIdxShift,
+                                                                   const Idx dimIndex = 0u) {
+      // Take into account the thread index in block.
+      const Idx threadIdxLocal(alpaka::getIdx<alpaka::Block, alpaka::Threads>(acc)[dimIndex]);
+      const Idx threadDimension(alpaka::getWorkDiv<alpaka::Thread, alpaka::Elems>(acc)[dimIndex]);
+
+      // Compute the elements indexes in block.
+      // Obviously relevant for CPU only.
+      // For GPU, threadDimension = 1, and elementIdx = firstElementIdx = threadIdx + elementIdxShift.
+      const Idx firstElementIdxLocal = threadIdxLocal * threadDimension;
+      const Idx firstElementIdx = firstElementIdxLocal + elementIdxShift;  // Add the shift!
+      const Idx endElementIdxUncut = firstElementIdx + threadDimension;
 
       // Return element indexes, shifted by elementIdxShift.
-      return {firstElementIdxVec, endElementIdxUncutVec};
+      return {firstElementIdx, endElementIdxUncut};
     }
 
     /*
      * Computes the range of the elements indexes, local to the block.
      * Truncated by the max number of elements of interest.
      */
-    template <typename T_Acc, typename T_Dim>
-    ALPAKA_FN_ACC std::pair<Vec<T_Dim>, Vec<T_Dim>> element_index_range_in_block_truncated(
-        const T_Acc& acc, const Vec<T_Dim>& maxNumberOfElements, const Vec<T_Dim>& elementIdxShift) {
+    template <typename TAcc>
+    ALPAKA_FN_ACC std::pair<Idx, Idx> element_index_range_in_block_truncated(const TAcc& acc,
+                                                                             const Idx maxNumberOfElements,
+                                                                             const Idx elementIdxShift,
+                                                                             const Idx dimIndex = 0u) {
       // Check dimension
-      static_assert(alpaka::Dim<T_Acc>::value == T_Dim::value,
-                    "Accelerator and maxNumberOfElements need to have same dimension.");
-      auto&& [firstElementIdxLocalVec, endElementIdxLocalVec] = element_index_range_in_block(acc, elementIdxShift);
+      //static_assert(alpaka::Dim<TAcc>::value == Dim1::value,
+      //              "Accelerator and maxNumberOfElements need to have same dimension.");
+      auto [firstElementIdxLocal, endElementIdxLocal] = element_index_range_in_block(acc, elementIdxShift, dimIndex);
 
       // Truncate
-      for (typename T_Dim::value_type dimIndex(0); dimIndex < T_Dim::value; ++dimIndex) {
-        endElementIdxLocalVec[dimIndex] = std::min(endElementIdxLocalVec[dimIndex], maxNumberOfElements[dimIndex]);
-      }
+      endElementIdxLocal = std::min(endElementIdxLocal, maxNumberOfElements);
```

Comment:
IIUC this makes sense only for 1D grids ?

Reply:
Yes.
I had done helpers targeting more dimensions at first, but then realized that the use cases of 2D grids for the pixel track reconstruction were very few, and that actually for these, I could not use the 2D helper functions as they were.
I had taken care that the `indexShift` / `maxNumberOfElements` could obviously be specified for each dimension, but there was the extra issue that one can do a grid stride for a given dimension, and a block stride for another dimension. Plus issues with the body of the lambda. Overall, this was making things more complicated then helpful on call site.

In legacy, the 2D grids are in `plugin-PixelTriplets/alpaka/gpuPixelDoubletsAlgos.h`, `plugin-PixelTriplets/alpaka/CAHitNtupletGeneratorKernelsImpl.h` and `plugin-PixelTriplets/alpaka/gpuFishbone.h`. I kept the use of 2D grids, just found it equivalent and much simpler to call 1D helpers for each dimension on call site (which is what the 2D helper would have been doing anyway).